### PR TITLE
Borrow integration

### PR DESCRIPTION
**Pull request by matzapata**
## Overview

Jira ticket: [SAF-11](https://wakeuplabs.atlassian.net/browse/SAF-11)

Borrow integration, calculations and transactions assembling.

## Comments

Includes changes from https://github.com/wakeuplabs-io/sovryn-dapp/pull/23 I found a bug there, so will fix that first or try to split the changes to allow for this merge if takes longer. Current contracts deployment fail for borrow action so in demo don't move it through but parameters and methods have been tested in polygon.

## Demo

https://github.com/user-attachments/assets/9ecb6aaf-3bd3-4690-addf-ab8a45888de2

### CODE REVIEW

**Review by luchopcerra — APPROVED** [review]
LGTM; 🚀

File: apps/frontend/src/app/5_pages/AavePage/components/BorrowAssetsList/components/BorrowModal/BorrowForm.tsx [hunk 1]
```diff
@@ -108,38 +135,56 @@ export const BorrowForm: FC<BorrowFormProps> = () => {
...
-          middleEnd={collateralRatioThresholds.MIDDLE_END}
-          end={collateralRatioThresholds.END}
-          value={collateralRatio.toNumber()}
+          start={0}
+          end={4 * 25} // maximum health rate for graph 4. So multiply all by 25 to reach 100
+          middleStart={1 * 25} // health factor < 1 => liquidation
+          middleEnd={1.2 * 25}
+          value={(newHealthFactor.toNumber() ?? 0) * 25}
         />
       </div>
```
Comment (matzapata) [inline apps/frontend/src/app/5_pages/AavePage/components/BorrowAssetsList/components/BorrowModal/BorrowForm.tsx]: There's a component for all this, CollateralRateHealthBar. I'll implement it

**Comment by matzapata** [comment]
Used mentioned component, replaced apy with apr and health factor with collateral ratio

Sorry, pretty awful video jejej

Uploading 2024-08-23 18-43-09.mp4…

**Comment by matzapata** [comment]
Used mentioned component, replaced apy with apr and health factor with collateral ratio

Sorry, pretty awful video jejej

https://github.com/user-attachments/assets/ebccb9c6-98d5-4107-a87f-563209914176

**Pull request merged**